HomeWork_12/HW12.py

```diff
@@ -0,0 +1,62 @@
+from datetime import datetime
+def timer():
+    start = datetime()
+    def inner():
+        return datetime() - start
+    return inner
+
+def funcname(func):
+    def inner(*args,**kwargs):
+        print(f"Функція {func.__name__}")
+        return func(*args,**kwargs)
+
+import json
+
+
+
+phonebook = {
+        "Aleks": "347638756",
+        "Fedor": "3784638754",
+        "Den": "2387463745"
+
+    }
+json_phonebook = json.dumps(phonebook)
+while True:
+    @timer
+    @funcname:
+    command = input('Enter a command: ')
+    if command == 'list':
+        for key in phonebook:
+            print(key)
+    elif command == 'stats':
+        L = phonebook
+        print(len(L))
+    elif command == 'show':
+        s = input("Enter a name: ")
+        if phonebook.get(s):
+            print(phonebook[s])
+    elif command == 'add':
+        n = input("Enter a name: ")
+        p = int(input("Enter a phone: "))
+
+        if n != phonebook:
+            phonebook[n] = p
+            with open("/Users/AleksandrPitelin/PythonProject/PythonProject/HomeWork_12/Json_hw12", "w") as file_Json:
```

Comment:
It would be better to use here relative path since the current one is valid only for your computer. Like: `open('Json_hw12, 'w') as file_Json`